src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -484,7 +477,9 @@ async def run_setup_pys(
         setup_py_results = await MultiGet(
```

Comment:
I don't think that interpreter constraints are calculated the way we want for the `setup-py` goal. Rather than looking at the transitive closure of each distinct `python_distribution`, we seem to be merging into one global constraint set? And only looking at the exported targets, rather than the closure.

@benjyw does that sound right? If so, that's likely a bug. Even though `setup-py` is deprecated, I should fix that in this PR.

Reply:
Yes, what you are saying appears to be correct, and this is a bug. It's working fine for the `package` goal though.